tripzone/src/containers/FlightSearch/FlightSearch.js

```diff
@@ -0,0 +1,151 @@
+import React, { useState, useEffect } from "react";
+import FlightService from "../../services/FlightService";
+import "./FlightSearch.scss";
+import Button from "../../components/Button/Button";
+import { getCities } from "../../services/TourismService";
+import logo from "../../assets/logo.png";
+import Dropdown from "../../components/Dropdown/Dropdown";
+import FlightCard from "../../components/Card/FlightCard/FlightCard";
+import BookCard from "../../components/Card/BookNowCard/BookCard";
+import { DestinationContext } from "../../screens/HomePage/HomePage";
+import { AppConstants } from "../../constants/App.constants";
+import { useContext } from "react";
+
+const FlightSearch = () => {
+  const { setDestination } = useContext(DestinationContext);
+  const [sourceCode, setSourceCode] = useState("");
+  const [destinationCode, setDestinationCode] = useState("");
+  const [flightData, setFlightData] = useState([]);
+  const [showBookCard, setShowBookCard] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const [cities, setCities] = useState([]);
+  const [selectedFlight, setSelectedFlight] = useState(null);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  const {FLIGHT_SEARCH} = AppConstants.COMMONS;
+ 
+  // Function used to handle when a city is Searched
+
+  const handleSearch = async (e) => {
+    e.preventDefault();
+    setIsLoading(true);
+    setError(null);
+    try {
+      const data = await FlightService(sourceCode, destinationCode);
+      setFlightData(data);
+      setIsSubmitted(true);
+      setShowBookCard(false);
```

Comment:
could you let me know why this flag should be changed to false

Reply:
Unless set to false, the book card will continue to display outdated data even after the destination has been changed. The card will only update when another "BOOK NOW" button is clicked for a different flight.